src/Plot.js

```diff
@@ -387,6 +387,14 @@ export default class Plot extends Viz {
     yDomain = y.domain();
     y2Domain = y2.domain();
 
+    const startTime = date(data[0].discrete);
+    const endTime = date(data[data.length - 1].discrete);
+
+    if (xTime) xDomain = [startTime, endTime];
+    if (x2Time) x2Domain = [startTime, endTime];
+    if (yTime) yDomain = [startTime, endTime];
+    if (y2Time) y2Domain = [startTime, endTime];
+
```

Comment:
these 6 lines aren't necessary. the domains are already parsed into times in the code above (`xDomain = xDomain.map(date);`)

Reply:
`xDomain = xDomain.map(date);` doesn't capture changes in Timeline, and Those lines reflect the new data, solving #106. 
![pr](https://user-images.githubusercontent.com/5233593/42237378-91bbd97c-7ecb-11e8-8a96-63e97ce75471.gif)

Reply:
I dug into this more, and it's actually the line below (`this._xDomain = xDomain;`) that is causing the problem! `this._xDomain` should only every be used for a user-defined xDomain, and the reason the `.map(date)` isn't working is because it's using the previous xDomain!

You should be able to delete these six lines, AND the line below, and it should all start working. Please test and let me know if it doesn't.